### PR TITLE
feat: DEV-3346: Task history is stored on the backend for the label stream

### PR DESCRIPTION
**Pull request by juliosgarbi**

### CODE REVIEW

File: src/sdk/lsf-sdk.js [hunk 1]
```diff
@@ -218,19 +218,56 @@ export class LSFWrapper {
...
+        _annotationId = taskHistory[0].previous.annotation || taskHistory[0].previous.draft;
+      }
+
       const newTask = await this.withinLoadingState(async () => {
-        if (!isDefined(taskID)) {
+        if (!isDefined(_taskId) && !isPrevious) {
           return tasks.loadNextTask();
         } else {
-          return tasks.loadTask(taskID);
+          return tasks.loadTask(_taskId);
```
Comment (bmartel) [inline src/sdk/lsf-sdk.js:270]: can we swap this for the affirmative so it reads a bit easier:

```suggestion
        if (isDefined(_taskId) && isPrevious) {
          return tasks.loadTask(_taskId);
        } else {
          return tasks.loadNextTask();
```

**Review by bmartel — CHANGES_REQUESTED** [review]

File: src/sdk/lsf-sdk.js [hunk 2]
```diff
@@ -264,10 +264,10 @@ export class LSFWrapper {
       }
 
       const newTask = await this.withinLoadingState(async () => {
-        if (!isDefined(_taskId) && !isPrevious) {
-          return tasks.loadNextTask();
-        } else {
+        if (isDefined(_taskId) || isPrevious) {
```
Comment (bmartel) [inline src/sdk/lsf-sdk.js:267]: I was going to ask, if the `_taskId` is not defined and `isPrevious` is `true`, is it the intention to call `loadNextTask`?

Comment (bmartel) [reply src/sdk/lsf-sdk.js:267]: As right now, you would be calling loadTask on `undefined` or `null` possibly.

**Review by bmartel — APPROVED** [review]

**Review by bmartel — APPROVED** [review]

**Comment by hlomzik** [comment]
Implemented with a little different approach in https://github.com/HumanSignal/dm2/pull/150

**Pull request closed without merging**